editor/components/post-featured-image/index.js

```diff
@@ -96,14 +97,22 @@ const applyConnect = connect(
 	}
 );
 
-const applyWithAPIData = withAPIData( ( { featuredImageId, postTypeName } ) => {
+const applyWithAPIData = withAPIData( ( { postTypeName } ) => {
 	return {
-		media: featuredImageId ? `/wp/v2/media/${ featuredImageId }` : undefined,
 		postType: postTypeName ? `/wp/v2/types/${ postTypeName }?context=edit` : undefined,
 	};
 } );
 
+const applyWithSelect = withSelect( ( select, { featuredImageId } ) => {
+	const { getMedia } = select( 'core' );
+
+	return {
+		image: featuredImageId ? getMedia( featuredImageId ) : null,
```

Comment:
You've changed the prop name from `media` to `image` in the mapping function, and it's resulting in the featured image never being displayed (the underlying component expects a `media` prop).